<?xml version='1.0'?>

<launch>

  <!-- global parameters -->
  <arg         name           = 'geotiff_map_file_path'
               default        = '$(find hector_geotiff)/maps'/>
  <arg         name           = 'car_name'
               default        = 'car_1'/>

  <!-- running on simulated time -->
  <param       name           = 'use_sim_time'
               value          = 'true'/>

  <!-- static transform between world and map -->
  <node        name           = 'world_map_link'
               pkg            = 'tf'
               type           = 'static_transform_publisher'
               args           = '1.0 0.0 0.0 0.0 0.0 0.0 world map 100'/>

  <!-- hector_mapping default parameters -->
  <include     file           = '$(find hector_mapping)/launch/mapping_default.launch'>
  <arg         name           = 'base_frame'
               default        = '$(arg car_name)_base_link'/>
  <arg         name           = 'odom_frame'
               default        = '$(arg car_name)_base_link'/>
  <arg         name           = 'scan_topic'
               default        = '$(arg car_name)/scan'/> </include>

  <!-- geotiff mapping node -->
  <include     file           = '$(find hector_geotiff)/launch/geotiff_mapper.launch'>
  <arg         name           = 'trajectory_source_frame_name'
               value          = 'scanmatcher_frame'/>
  <arg         name           = 'map_file_path'
               value          = '$(arg geotiff_map_file_path)'/> </include>

  <!-- launch rviz with mapping configuration -->
  <node        pkg            = 'rviz'
               type           = 'rviz'
               name           = 'rviz'
               args           = '-d $(find f1tenth-sim)/rviz/mapping.rviz'/>

</launch>
